src/c++/formatter.cpp

```diff
@@ -45,7 +45,12 @@ meto::Formatter::Formatter() {
  * @param[in] hashvec  Vector of data that the format method will operate on
  */
 
-void meto::Formatter::execute_format(std::ostream &os, hashvec_t hashvec) {
+void meto::Formatter::execute_format(std::ostream &os, hashvec_t hashvec,
+                                     MPIContext &mpi_context) {
+  // Add an MPI task identifier to each output file
+  os << "\n"
+     << "MPI task " << mpi_context.get_rank() << " of "
+     << mpi_context.get_size() << "\n";
```

Comment:
The output here comes out as `MPI task 0 of 1` I'm wondering whether it is better to have `MPI task 1 of 1 : rank ID 0` this would give both the integer value of the total whilst also having the rank id without the confusion of the total number being offset by one.

Reply:
That's a simpler alternative (or at least simpler to read if not to implement 😁)

Reply:
That would work well, it would make it clearer in the multi file output as well as each file would still tell you how many ranks there are in total.